### PR TITLE
Development changes made in #194

### PR DESCRIPTION
**Pull request by DonIsaac**
Extracted improvements to `Gemfile`, the various `docker-compose` configs, and `openapi.yaml` from #194 while simultaneously implementing the requested changes

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -17,11 +17,14 @@ group :development do
   gem 'rspec'
   gem 'shotgun'
   gem 'solargraph'
+  gem 'debase'
+  gem 'ruby-debug-ide', require: false
```
Comment (tybug) [inline Gemfile:21]: I have yet to really understand ruby's whole dependency and bundler ecosystem, but why isn't `debase` also `require: false`? does our actual code call it anywhere? shouldn't it be not required like ruby-debug-ide

File: Gemfile [hunk 2]
```diff
@@ -17,11 +17,14 @@ group :development do
...
 end
 
 group :test do
   gem 'rack-test', require: 'rack/test'
   gem 'simplecov', require: false
+  gem 'json-schema'
```
Comment (tybug) [inline Gemfile:27]: where is this gem actually being used?

**Review by McIntireEvan — APPROVED** [review]
thanks for splitting it!

File: openapi.yaml [hunk 3]
```diff
@@ -36,15 +36,31 @@ servers:
   - url: "https://api.umd.io/v1"
 tags:
   - name: courses
-    description: This set of endpoints lets you get data about university courses and their sections. You can get one course or section at a time, several courses or sections at a time, or a list of all the courses. All the data is scraped from Testudo.
+    description: >-
+      This set of endpoints lets you get data about university courses and their
+      sections. You can get one course or section at a time, several courses or
+      sections at a time, or a list of all the courses. All the data is scraped
+      from Testudo.
+
```
Comment (tybug) [inline openapi.yaml:44]: a general comment about the line splitting occurring in this file: is there a standard char limit per line for yaml files? If not, should we enforce one of our own? I'm definitely a fan of char limits (I follow python's 80 char limit pretty religiously, though most people aren't happy with it being that short) so I wouldn't be against enforcing our own, even if only colloquially between us (we all set a ruler in vscode at 100 chars or something).

**Review by tybug — CHANGES_REQUESTED** [review]

File: openapi.yaml (hunk 3 above)
Comment (DonIsaac) [reply openapi.yaml:44]: I generally stick to 80 characters. Also, the `>-` means the final string doesn't occur a newline (like they're not preserved) so
newlines here are just aesthetic.

File: Gemfile (hunk 1 above)
Comment (DonIsaac) [reply Gemfile:21]: Didn't include it because that's not how they said it in their readme. I think it needs to not be there

File: Gemfile (hunk 2 above)
Comment (DonIsaac) [reply Gemfile:27]: nowhere, really, was planning on using it for testing. I'll remove it.

**Pull request merged**